basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java

```diff
@@ -484,6 +484,22 @@ public String buildToolSitePanelContext(VelocityPortlet portlet, Context context
 					log.error("error getting url for site {}", siteId);
 				}
 
+				// Patch the date type
+				// https://stackoverflow.com/questions/19431234/converting-between-java-time-localdatetime-and-java-util-date
+				Object created_at = content.get("created_at");
+				if ( created_at instanceof java.util.Date ) {
+			        DateFormat df2 = DateFormat.getDateTimeInstance(DateFormat.MEDIUM, DateFormat.MEDIUM, rb.getLocale());
+					String output = df2.format(created_at);
+					content.put("created_at", output);
+				} else if ( created_at instanceof java.time.LocalDateTime) {
+					java.time.LocalDateTime ldt = (java.time.LocalDateTime) created_at;
+					String output = java.time.format.DateTimeFormatter.ofLocalizedDate(java.time.format.FormatStyle.MEDIUM).withLocale(rb.getLocale()).format(ldt.now());
```

Comment:
The UserTimeService does a lot of this conversion for you, and it uses the users preferred TZ.

Reply:
Agree with @ern above. This is incorrect as is because it is not localizing to user preferred timezone.

UserTimeService will do all this for you in user's locale with their preferred timezone

./kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java

Reply:
Thanks - I figured there was a better way.